src/Peachpie.Library/OpenSSL.cs

```diff
@@ -0,0 +1,741 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Pchp.Core;
+using System.Security.Cryptography;
+using System.IO;
+using static Pchp.Library.PhpHash;
+using System.Security.Cryptography.X509Certificates;
+using Pchp.Core.Utilities;
+
+namespace Pchp.Library
+{
+    /// <summary>
+    /// PHP openssl support.
+    /// </summary>
+    [PhpExtension("openssl")]
+    public static class OpenSSL
+    {
+        #region Constants
+        public const int OPENSSL_RAW_DATA = (int)Options.OPENSSL_RAW_DATA;
+        public const int OPENSSL_ZERO_PADDING = (int)Options.OPENSSL_ZERO_PADDING;
+
+        private static Dictionary<string, Cipher> Ciphers = new Dictionary<string, Cipher>
+        {
+            {"aes-256-cbc", new Cipher(CipherType.AES, Cipher.IVLengthAES, CipherMode.CBC,256)},
+            {"aes-192-cbc", new Cipher(CipherType.AES, Cipher.IVLengthAES, CipherMode.CBC, 192)},
+            {"aes-128-cbc", new Cipher(CipherType.AES, Cipher.IVLengthAES, CipherMode.CBC,128)},
+            {"aes-256-ecb", new Cipher(CipherType.AES, Cipher.IVLengthAES, CipherMode.ECB, 256)},
+            {"aes-192-ecb", new Cipher(CipherType.AES, Cipher.IVLengthAES, CipherMode.ECB,192)},
+            {"aes-128-ecb", new Cipher(CipherType.AES, Cipher.IVLengthAES, CipherMode.ECB, 128)},
+            {"des-ecb", new Cipher(CipherType.DES, Cipher.IVLengthDES, CipherMode.ECB, Cipher.KeyLengthDES)},
+            {"des-cbc", new Cipher(CipherType.DES, Cipher.IVLengthDES,CipherMode.CBC, Cipher.KeyLengthDES)},
+            {"des-ede3", new Cipher(CipherType.TripleDES, Cipher.IVLengthDES, CipherMode.ECB, Cipher.KeyLengthTripleDES)},
+            {"des-ede3-cbc", new Cipher(CipherType.TripleDES, Cipher.IVLengthDES, CipherMode.CBC, Cipher.KeyLengthTripleDES)}
+        // CFB mode is not supported in .NET Core yet https://github.com/dotnet/runtime/issues/15771
+        // RC2 is ok when there is right length of password, but when it is longer, PHP transforms password in some way, but i can not figure out how.
+        // Parameters tag and add are for gcm and ccm cipher mode. (I found implementation in version .Net Core 3.0 and 3.1)
+        };
+
+        private static Dictionary<string, string> CiphersAliases = new Dictionary<string, string>
```

Comment:
is it case sensitive or insensitive?

Reply:
Yes, it is. I have already corrected it.

Reply:
I see :) thanks

Suggestion for the next time - instead of calling `.ToLower()` everywhere, just create the Dictionary itself case-insensitive. `ToLower()` has an unnecessary overhead of getting current CultureInfo and allocating the new string